python/ray/tune/integration/xgboost.py

```diff
@@ -51,33 +64,42 @@ class TuneReportCallback(TuneCallback):
     .. code-block:: python
 
         import xgboost
-        from ray.tune.integration.xgboost import TuneReportCallback
+        from ray.tune.integration.xgboost import TuneReportCheckpointCallback
 
         config = {
             # ...
             "eval_metric": ["auc", "logloss"]
         }
 
-        # Report only log loss to Tune after each validation epoch:
+        # Report only log loss to Tune after each validation epoch.
+        # Save model as `xgboost.mdl`.
         bst = xgb.train(
             config,
             train_set,
             evals=[(test_set, "eval")],
             verbose_eval=False,
-            callbacks=[TuneReportCallback({"loss": "eval-logloss"})])
+            callbacks=[TuneReportCheckpointCallback(
+                {"loss": "eval-logloss"}, "xgboost.mdl)])
 
     """
 
+    _checkpoint_callback_cls = None
+    _report_callbacks_cls = None
```

Comment:
Remove these?

Reply:
We check if these properties are set for backwards compatibility with xgboost-ray and lightgbm-ray (and also for sequencing the API update).

We could test this also with `getattr` but it makes the code that invokes them more cluttered.